.github/workflows/ci-tests.yml

```diff
@@ -26,13 +26,20 @@ jobs:
     strategy:
       matrix:
         jobs: [ 'linting', 'doclinting', 'docbuild' ]
+    env:
+      PYTHON_VERSION: '3.10'
     name: ${{ matrix.jobs }} tests
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python
       uses: actions/setup-python@v2
       with:
-        python-version: '3.10'
+        python-version: ${{ env.PYTHON_VERSION }}
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.pythonLocation }}
+        key: ${{ env.PYTHON_VERSION }}-${{ runner.os }}-build-${{ hashFiles('tox.ini') }}
```

Comment:
these static linting packages should really be installed in their relevant tox section, to avoid needlessly installing in the unit tests. So feel free to make that change and keep the hash as is if you want. If not we can do that later :)